dali/operators/random/noise/salt_and_pepper_noise.h

```diff
@@ -38,12 +38,14 @@ class SaltAndPepperNoiseImpl {
       std::is_floating_point<T>::value ? T(-1) : std::numeric_limits<T>::min();
 
   DALI_HOST_DEV explicit SaltAndPepperNoiseImpl(float noise_prob = 0.05,
-                                                float salt_to_pepper_prob = 0.5,
+                                                float salt_vs_pepper = 0.5,
                                                 T salt_val = kDefaultSalt,
                                                 T pepper_val = kDefaultPepper)
-      : noise_prob_(clamp<float>(noise_prob, 0, 1)),
-        salt_prob_(noise_prob_ * clamp<float>(salt_to_pepper_prob, 0, 1)),
+      : noise_prob_(noise_prob),
+        salt_prob_(noise_prob_ * salt_vs_pepper),
         salt_val_(salt_val), pepper_val_(pepper_val) {
+      assert(0.0f <= noise_prob && noise_prob <= 1.0);
+      assert(0.0f <= salt_vs_pepper && salt_vs_pepper <= 1.0);
```

Comment:
Please remove this from device code. Asserts cost, sometimes a lot. Do you even know if this works in debug mode without exceeding launch resources?

Reply:
I've tested it in debug mode. Anyway, removed the assert.